application/common/dos.py

```diff
@@ -115,15 +115,21 @@ def get_matching_dos_services(odscode: str, org_type_id: str) -> List[DoSService
         list[DoSService]: List of DoSService objects with matching first 5
         digits of odscode, taken from DoS database
     """
-    odscode = get_new_odscode_for_dos(odscode, org_type_id)
     logger.info(f"Searching for '{org_type_id}' DoS services with ODSCode that matches '{odscode}'")
+
+    if org_type_id == PHARMACY_ORG_TYPE_ID:
+        odscode_for_sql = odscode[0:5]
+    elif org_type_id == DENTIST_ORG_TYPE_ID:
+        odscode_for_sql = odscode[0:1] + odscode[2:]
+    else:
+        odscode_for_sql = odscode
     sql_query = (
         "SELECT s.id, uid, s.name, odscode, address, town, postcode, web, email, fax, nonpublicphone, typeid,"
         " parentid, subregionid, statusid, createdtime, modifiedtime, publicphone, publicname, st.name servicename"
         " FROM services s LEFT JOIN servicetypes st ON s.typeid = st.id"
         " WHERE odscode LIKE %(ODS)s"
```

Comment:
I can't see how you are connecting dentists odscode of length 7 to DoS odscode of length 6. This scenario covers the matching length 7 from NHS UK to matching DoS lengths 7 or larger correctly.

Reply:
The pharmacy and dentist will likely need different where statements

Reply:
odscode = "V012345" # 7 chars 
print (odscode[0:1] + odscode[2:]) #res V12345 6 chars